code/game/gamemodes/clock_cult/clock_items.dm

```diff
@@ -597,20 +659,33 @@
 	if(stored_alloy - alloy_cost < 0)
 		user << "<span class='warning'>You need [alloy_cost] replicant alloy to proselytize [target]!</span>"
 		return 0
+	if(stored_alloy - alloy_cost > 100)
+		user << "<span class='warning'>You have too much replicant alloy stored to proselytize [target]!</span>"
+		return 0
+	if(ratvar_awakens) //Ratvar makes it faster
+		operation_time *= 0.5
 	user.visible_message("<span class='warning'>[user]'s [src] begins tearing apart [target]!</span>", "<span class='brass'>You begin proselytizing [target]...</span>")
 	playsound(target, 'sound/machines/click.ogg', 50, 1)
-	if(!do_after(user, operation_time, target = target))
+	if(operation_time && !do_after(user, operation_time, target = target))
 		return 0
 	if(stored_alloy - alloy_cost < 0) //Check again to prevent bypassing via spamclick
 		return 0
+	if(stored_alloy - alloy_cost > 100)
+		return 0
+	if(!target || target.type != target_type)
```

Comment:
If the target has been deleted or is a different type than what it was before, we cancel; this would mostly happen due to spamclicking.